pulsar-client-cpp/python/pulsar/functions/context.py

```diff
@@ -126,7 +126,8 @@ def record_metric(self, metric_name, metric_value):
   @abstractmethod
   def publish(self, topic_name, message, serde_class_name="serde.IdentitySerDe", properties=None, compression_type=None, callback=None):
```

Comment:
I am not a python expert. but this PR doesn't seem to specify `partition_key` in the method signature. how does that work?

Reply:
@sijie I have updated the implementation

Reply:
can we add a ut or integration test for this? this raises a huge concern to me - we update the implementation but didn't change the interface definition. how can we guarantee the fix is correct?

Reply:
@sijie you reviewed the PR when I was in the middle of updating the PR due to discussions and comments around this PR.  The original PR did not modify any of the interfaces, but after a discussion with @srkukarni and @merlimat, we decided it was best if we let the user choose what the partition key is.  That said, I will add some tests to it

Reply:
@sijie if you look at the commit you commented on.  The signature of contextimpl.publish didn't change

Reply:
Sure I might be commenting in between commits. but my question is still there : how can we guarantee the fix is correct? where are the tests for this change?